packages/app/src/specs/CreateSpecModal.vue

```diff
@@ -105,7 +111,15 @@ const specFileName = computed(() => {
   return getPathForPlatform(props.gql.currentProject?.defaultSpecFileName || '')
 })
 
-const filteredGenerators = getFilteredGeneratorList(props.gql.currentProject?.currentTestingType)
+const codeGenGlob = computed(() => {
+  if (!generator.value) {
+    return null
+  }
+
+  return props.gql.currentProject?.codeGenGlobs[generator.value.id]
```

Comment:
Being unfamiliar with a lot of the `generator` design, this line threw me for a loop. Wasn't quite sure what was being looked up here. I eventually worked it out, but digging in left me wondering why we're deriving this here and passing it as a prop to the generator component? It seems like the generator component could determine this itself in a direct fashion, as it implicitly knows which type of generator it is, and leave the CreateSpecModal with less to care about. 

The same might be true for `specFileName` as well.

Reply:
Yeah that was unnecessarily confusing. I updated it to just get whichever `codeGenGlob` it needs from `gql.currentProject`